docs/source/schema-directives.md

```diff
@@ -1,179 +1,581 @@
 ---
 title: Schema directives
-description: Using custom Directive on FIELD_DEFINITION
+description: Implementing and using custom `@directive`s that transform schema types, fields, and arguments
 ---
 
-## Simple Directive example
+A _directive_ is an identifier preceded by a `@` character, optionally followed by a list of named arguments, which can appear after almost any form of syntax in the GraphQL query or schema languages. Here's an example from the [GraphQL draft specification](http://facebook.github.io/graphql/draft/#sec-Type-System.Directives) that illustrates several of these possibilities:
 
-Let's take a look at how we can create `@upper` Directive to upper-case a string returned from resolve on Field
-[See a complete runnable example on Launchpad.](https://launchpad.graphql.com/p00rw37qx0)
+```typescript
+directive @deprecated(
+  reason: String = "No longer supported"
+) on FIELD_DEFINITION | ENUM_VALUE
 
-To start, let's grab the schema definition string from the `makeExecutableSchema` example [in the "Generating a schema" article](/tools/graphql-tools/generate-schema.html#example).
+type ExampleType {
+  newField: String
+  oldField: String @deprecated(reason: "Use `newField`.")
+}
+```
+
+Points to notice in this example:
+
+* The `@deprecated` directive _follows_ the field that it pertains to (`oldField`), even though its syntax might remind you of "decorators" in other languages, which usually appear on the line above.
+* Directives are often _declared_ in the schema, as in this example, though it's up to the GraphQL server to enforce the argument types (`reason: String`) and locations (`FIELD_DEFINITION | ENUM_VALUE`) of the declaration.
+* The `@deprecated(reason: ...)` syntax is legal even without the `directive @deprecated ...` declaration.
```

Comment:
Personally I find this confusing - maybe we can just have some advanced section about defining directives at the bottom of the page? It's weird that defining them is optional and maybe we should take a stance on whether or not people should do it.